File: src/main/java/routing_game/Model/RoutingEntry.java
package routing_game.Model;

import java.util.Objects;

public final class RoutingEntry {
    private final int destination;
    private final int nextHop;
    private final int cost;

    public RoutingEntry(int destination, int nextHop) {
        this(destination, nextHop, 0);
    }

    public RoutingEntry(int destination, int nextHop, int cost) {
        this.destination = destination;
        this.nextHop = nextHop;
        this.cost = cost;
    }

    public int destination() {
        return destination;
    }

    public int nextHop() {
        return nextHop;
    }

    public int cost() {
        return cost;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof RoutingEntry other)) {
            return false;
        }
        return destination == other.destination && nextHop == other.nextHop;
    }

    @Override
    public int hashCode() {
        return Objects.hash(destination, nextHop);
    }
}

File: web/src/model/RoutingEntry.js
export class RoutingEntry {
  constructor(destination, nextHop, cost = 0) {
    this._destination = destination;
    this._nextHop = nextHop;
    this._cost = cost;
  }

  destination() {
    return this._destination;
  }

  nextHop() {
    return this._nextHop;
  }

  cost() {
    return this._cost;
  }

  equals(other) {
    return other instanceof RoutingEntry
      && this._destination === other._destination
      && this._nextHop === other._nextHop;
  }
}
